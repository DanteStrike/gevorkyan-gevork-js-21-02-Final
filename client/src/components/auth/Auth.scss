@import "src/assets/variables";

.auth {
  display: flex;
  align-items: center;
  justify-content: flex-end;
  width: 180px;
  height: 50px;

  &__link {
    padding: 0 0 0 8px;
    font-family: var(--secondary-font-family);
    font-size: 18px;
    color: inherit;
  }

  &__link--left {
    padding-right: 8px;
    border-right: var(--primary-border);
  }

  &__link--small {
    font-size: 12px;
    color: var(--secondory-font-color);
  }

  &__btn {
    padding: 4px 0 0 8px;
    font-family: var(--secondary-font-family);
    font-size: 12px;
    color: var(--secondory-font-color);
    cursor: pointer;
    background: unset;
    border: none;
    transition: color 0.3s;
  }

  &__btn:hover {
    color: #1890ff;
  }

  &__link--profile {
    display: flex;
    flex-grow: 1;
    align-items: center;
    justify-content: flex-end;
  }

  &__user {
    max-width: 108px;
    margin-left: 5px;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }
}

@media (max-width: $tabletWidth) {
  .auth {
    justify-content: center;
    height: 40px;

    &__link {
      padding: 0 8px;
      font-size: 16px;
    }
  }
}
