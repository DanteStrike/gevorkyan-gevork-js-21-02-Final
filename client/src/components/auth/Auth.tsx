import React from 'react';
import './Auth.scss';
import {Avatar} from 'antd';
import CustomLink from '../custom-link/CustomLink';
import {RoutePath} from '../../enums';
import {RouteUtils} from '../../utils';
import useAppTranslation from '../../hooks/use-app-translation';

interface IAuthProps {
  isAuth: boolean;
  onLogout: () => void;
  authUser?: {
    id: string;
    name: string;
    picture: string;
  };
}

function Auth({
  isAuth,
  onLogout,
  authUser = {
    id: ``,
    name: ``,
    picture: ``,
  },
}: IAuthProps) {
  const {t} = useAppTranslation(`auth`);
  const {name, picture, id} = authUser;
  const profileRoute = RouteUtils.createProfileRoute(id);

  return (
    <div className="auth">
      {!isAuth ? (
        <>
          <CustomLink className="auth__link auth__link--left" to={RoutePath.LOGIN}>
            {t(`login`)}
          </CustomLink>
          <CustomLink className="auth__link" to={RoutePath.REGISTRATION}>
            {t(`registration`)}
          </CustomLink>
        </>
      ) : (
        <>
          <CustomLink className="auth__link auth__link--left auth__link--profile" to={profileRoute}>
            <Avatar src={picture} />
            <span className="auth__user">{name}</span>
          </CustomLink>
          <button type="button" className="auth__btn" onClick={onLogout}>
            {t(`logout`)}
          </button>
        </>
      )}
    </div>
  );
}

Auth.defaultProps = {
  authUser: {
    id: ``,
    name: ``,
    picture: ``,
  },
};

export default React.memo(Auth);
