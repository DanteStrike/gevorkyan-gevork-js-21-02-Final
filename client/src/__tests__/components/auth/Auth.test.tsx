import {shallow} from 'enzyme';
import React from 'react';
import Auth from '../../../components/auth/Auth';

describe(`Component Comment should render correctly`, () => {
  it(`snapshot default test`, () => {
    const wrapper = shallow(<Auth isAuth={false} onLogout={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });

  it(`snapshot isAuth test`, () => {
    const wrapper = shallow(
      <Auth
        isAuth
        authUser={{
          id: `id`,
          name: `name`,
          picture: `url`,
        }}
        onLogout={jest.fn()}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
