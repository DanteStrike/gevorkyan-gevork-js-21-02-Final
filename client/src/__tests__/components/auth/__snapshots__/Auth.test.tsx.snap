// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Comment should render correctly snapshot default test 1`] = `
<div
  className="auth"
>
  <Memo(CustomLink)
    className="auth__link auth__link--left"
    to="/login"
  >
    Login
  </Memo(CustomLink)>
  <Memo(CustomLink)
    className="auth__link"
    to="/registration"
  >
    Registration
  </Memo(CustomLink)>
</div>
`;

exports[`Component Comment should render correctly snapshot isAuth test 1`] = `
<div
  className="auth"
>
  <Memo(CustomLink)
    className="auth__link auth__link--left auth__link--profile"
    to="/profile/id"
  >
    <Avatar
      shape="circle"
      size="default"
      src="url"
    />
    <span
      className="auth__user"
    >
      name
    </span>
  </Memo(CustomLink)>
  <button
    className="auth__btn"
    onClick={[MockFunction]}
    type="button"
  >
    Logout
  </button>
</div>
`;
